Sources/KSCrashRecording/Monitors/KSCrashMonitor_Memory.h

```diff
@@ -0,0 +1,94 @@
+//
+//  KSCrashMonitor_Memory.h
+//
+//  Copyright (c) 2024 Alexander Cohen. All rights reserved.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall remain in place
+// in this source code.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+
+
+/* Monitor memory and records data for OOMs.
+ */
+
+
+#ifndef KSCrashMonitor_Memory_h
+#define KSCrashMonitor_Memory_h
+
+#include "KSCrashMonitor.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/** States of transition for the application */
+enum {
+    KSCrash_ApplicationTransitionStateNone = 0,
+    KSCrash_ApplicationTransitionStateLaunching,
+    KSCrash_ApplicationTransitionStateForegrounding,
+    KSCrash_ApplicationTransitionStateActive,
+    KSCrash_ApplicationTransitionStateDeactivating,
+    KSCrash_ApplicationTransitionStateBackground,
+    KSCrash_ApplicationTransitionStateTerminating,
+};
+typedef uint8_t KSCrash_ApplicationTransitionState;
+
+/** App Memory */
+typedef struct KSCrash_Memory {
```

Comment:
As this struct is mapped to a file, we should ensure binary compatibility and handle padding and alignment.

Reply:
Come to think of it, this is a single device only format which will always be loaded from the same device it was saved on. Because of that, we don't really need to handle any of this.

Reply:
Actually I was thinking of `__attribute__((packed))`.

Reply:
> Actually I was thinking of `__attribute__((packed))`.

Same here. I was using `#pragma push(pack, 1)` but same/same. Point is, it'll simply make the structure a weird size and possible align things a bit oddly which could make the reads/writes slower.

Reply:
Oh, you want to avoid using alignment to improve performance. In that case, I’m okay with it.